src/repository.c

```diff
@@ -1066,25 +1066,42 @@ void git_repository_set_config(git_repository *repo, git_config *config)
 	set_config(repo, config);
 }
 
-int git_repository_odb__weakptr(git_odb **out, git_repository *repo)
+static int git_repository__setup_odb(git_odb **out, git_repository *repo)
 {
+	git_buf odb_path = GIT_BUF_INIT;
+	git_odb *odb;
 	int error = 0;
 
-	assert(repo && out);
+	if ((error = git_repository_item_path(&odb_path, repo,
+			GIT_REPOSITORY_ITEM_OBJECTS)) < 0 ||
+		(error = git_odb_new(&odb)) < 0)
```

Comment:
I know you're just using the preexisting code, but we actually have a memory leak here whenever `git_odb_new` errors. I mean in that case we probably have other problems because that indicates memory allocation errors, but still. Would be cool if you could refactor this a bit to not have multiple statements in a single if-condition and use `goto cleanup`-style to avoid those memory leaks.